app/approve/registrations/approve.user.registration/approve.user.registration.modal.controller.js

```diff
@@ -0,0 +1,188 @@
+(() => {
+
+    'use strict';
+
+    angular
+        .module('app')
+        .controller('ApproveUserRegistrationModalController', ApproveUserRegistrationModalController);
+
+    ApproveUserRegistrationModalController.$inject = ['$log', 'growl', '$uibModalInstance', 'currentUser', 'UserFactory', 'PublicInfoFactory', '$routeSegment', 'Ranks'];
+    function ApproveUserRegistrationModalController ($log, growl, $uibModalInstance, currentUser, UserFactory, PublicInfoFactory, $routeSegment, Ranks) {
+        const vm = this;
+        const confirmDeleteModal = '#ConfirmDelete';
+
+        vm.currentUser = currentUser;
+
+        vm.changeCountry = changeCountry;
+        vm.changeRegion = changeRegion;
+        vm.changeCity = changeCity;
+        vm.changePlatoon = changePlatoon;
+        vm.setSections = setSections;
+        vm.approveUser = approveCurrentUser;
+        vm.declineUser = declineCurrentUser;
+        vm.showConfirmDecline = showConfirmDecline;
+        vm.close = close;
+
+        activate();
+
+        ///
+
+        function activate() {
+            setCountries();
+            setRegions(vm.currentUser.country.id);
+            setCities(vm.currentUser.region.id);
+            setPlatoons(vm.currentUser.city.id);
+            setSections(vm.currentUser.platoon.id);
+            setRanks();
+        }
+
+        function approveCurrentUser() {
+            close();
```

Comment:
We must close modal window only in callback after success response. So invoke `close()` method in `approveRegistrationUser` success callback.